config/config.go

```diff
@@ -1086,6 +1096,47 @@ func (c *AzureSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return checkOverflow(c.XXX, "azure_sd_config")
 }
 
+// TritonSDConfig is the configuration for Triton based service discovery.
+type TritonSDConfig struct {
+	Account            string         `yaml:"account"`
+	Cert               string         `yaml:"cert"`
+	DnsSuffix          string         `yaml:"dns_suffix"`
+	Endpoint           string         `yaml:"endpoint"`
+	InsecureSkipVerify bool           `yaml:"insecure_skip_verify"`
```

Comment:
Anything speaking against using our standard TLS config object? https://github.com/prometheus/prometheus/blob/master/config/config.go#L963

We also have a helper to generate an HTTP client for it already: https://github.com/prometheus/prometheus/blob/master/util/httputil/client.go#L122-L152